publication-event-handlers/src/test/java/no/unit/nva/publication/events/handlers/batch/UpdatePublicationsInBatchesHandlerTest.java

```diff
@@ -97,10 +103,73 @@ void shouldNotUpdatePublicationPublisherWhenPublicationsHaveDifferentPublisherTh
         });
     }
 
-    private static InputStream createEvent(String publisherIdentifier, String newPublisherIdentifier) {
-        return IoUtils.stringToStream(new ManuallyUpdatePublicationsRequest(publisherIdentifier, newPublisherIdentifier,
+    @Test
+    void shouldUpdateFileLicenseUriWhenUpdateTypeIsLicenseAndLicenseUriIsProvidedInRequestAndMatchesFileLicense()
+        throws IOException {
+        var license = randomUri();
+        var newLicense = randomUri();
+        var publicationsToUpdate = createMultiplePublicationsWithLicense(license);
+        var event = createEvent(ManualUpdateType.LICENSE, license.toString(), newLicense.toString());
+
+        mockSearchApiResponseWithPublications(publicationsToUpdate);
+
+        handler.handleRequest(event, output, CONTEXT);
+
+        publicationsToUpdate.forEach(publication -> {
+            var updatedPublication = getPublicationByIdentifier(publication);
+
+            assertEquals(publication.getAssociatedArtifacts().size(),
+                         updatedPublication.getAssociatedArtifacts().size());
+
+            var updatedFiles = getFiles(updatedPublication);
+
+            updatedFiles.forEach(file -> assertEquals(newLicense, file.getLicense()));
+        });
+    }
+
+    @Test
+    void shouldNotUpdateFileLicenseUriWhenUpdateTypeIsLicenseAndLicenseUriIsNotEqualProvidedInRequestLicense()
+        throws IOException {
+        var license = randomUri();
+        var publicationsToUpdate = createMultiplePublicationsWithLicense(license);
+        var event = createEvent(ManualUpdateType.LICENSE, randomString(), randomString());
+
+        mockSearchApiResponseWithPublications(publicationsToUpdate);
+
+        handler.handleRequest(event, output, CONTEXT);
+
+        publicationsToUpdate.forEach(publication -> {
+            var updatedPublication = getPublicationByIdentifier(publication);
+
+            assertEquals(publication.getAssociatedArtifacts().size(),
+                         updatedPublication.getAssociatedArtifacts().size());
+
+            var updatedFiles = getFiles(updatedPublication);
+
+            updatedFiles.forEach(file -> assertEquals(license, file.getLicense()));
+        });
+    }
+
+    @Test
+    void shouldThrowIllegalArgumentExceptionWhenUnknownUpdateType() {
```

Comment:
This will fail due to the change made to `ManuallyUpdatePublicationUtil.java`, however, validate the creation of the enum object from JSON